ui/hooks/useName.test.ts

```diff
@@ -95,17 +112,75 @@ describe('useName', () => {
             name: NAME_MOCK,
             proposedNames: PROPOSED_NAMES_MOCK,
             sourceId: SOURCE_ID_MOCK,
+            origin: ORIGIN_MOCK,
           },
         },
       },
     });
 
     const nameEntry = useName(VALUE_MOCK, TYPE_MOCK, CHAIN_ID_2_MOCK);
 
-    expect(nameEntry).toStrictEqual({
+    expect(nameEntry).toStrictEqual<NameEntry>({
       name: NAME_MOCK,
       sourceId: SOURCE_ID_MOCK,
       proposedNames: PROPOSED_NAMES_MOCK,
+      origin: ORIGIN_MOCK,
+    });
+  });
+
+  describe('fallback variation', () => {
+    it('gets used if specified variation has no entry.', () => {
+      getNamesMock.mockReturnValue({
+        [TYPE_MOCK]: {
+          [VALUE_MOCK]: {
+            [FALLBACK_VARIATION]: {
+              name: NAME_MOCK,
+              proposedNames: PROPOSED_NAMES_MOCK,
+              sourceId: SOURCE_ID_MOCK,
+              origin: ORIGIN_MOCK,
+            },
+          },
+        },
+      });
+
+      const nameEntry = useName(VALUE_MOCK, TYPE_MOCK, CHAIN_ID_2_MOCK);
+
+      expect(nameEntry).toStrictEqual<NameEntry>({
+        name: NAME_MOCK,
+        sourceId: SOURCE_ID_MOCK,
+        proposedNames: PROPOSED_NAMES_MOCK,
+        origin: ORIGIN_MOCK,
+      });
+    });
+
+    it('gets used if specified variation has cleared name.', () => {
```

Comment:
Super minor, but our standard is no `.` at the end of test names.

Reply:
Done.